src/gluonts/model/n_beats/_network.py

```diff
@@ -0,0 +1,531 @@
+    num_block_layers
+        Number of fully connected layers with ReLu activation.
+    expansion_coefficient_length
+        If the type is "G" (generic), then the length of the expansion coefficient.
+        If type is "T" (trend), then it corresponds to the degree of the polynomial.
+        If the type is "S" (seasonal) then its not used.
+    block_type
+        Either "G" (generic), "S" (seasonal) or "T" (trend).
+    prediction_length
+        Length of the prediction. Also known as 'horizon'.
+    context_length
+        Number of time units that condition the predictions
+        Also known as 'lookback period'.
+    has_backcast
+        Only the last block of the network doesnt.
+    kwargs
+        Arguments passed to 'HybridBlock'.
+    """
+
+    # Needs the validated decorator so that arguments types are checked and
+    # the block can be serialized.
+    @validated()
+    def __init__(
+        self,
+        width: int,
+        num_block_layers: int,
+        expansion_coefficient_length: int,
+        prediction_length: int,
+        context_length: int,
+        block_type: str,
+        has_backcast: bool,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.width = width
+        self.num_block_layers = num_block_layers
+        self.expansion_coefficient_length = expansion_coefficient_length
+        self.block_type = block_type
+        self.prediction_length = prediction_length
+        self.context_length = context_length
+        self.has_backcast = has_backcast
+
+        with self.name_scope():
+            self.fc_stack = mx.gluon.nn.HybridSequential()
+            for i in range(num_block_layers):
+                self.fc_stack.add(
+                    mx.gluon.nn.Dense(units=self.width, activation="relu")
+                )
+            if block_type == "G":
+                if has_backcast:
+                    self.theta_backcast = mx.gluon.nn.Dense(
+                        units=expansion_coefficient_length  # linear activation:
+                    )
+                    self.backcast = mx.gluon.nn.Dense(
+                        units=context_length  # linear activation:
+                    )
+                self.theta_forecast = mx.gluon.nn.Dense(
+                    units=expansion_coefficient_length  # linear activation:
+                )
+                self.forecast = mx.gluon.nn.Dense(
+                    units=prediction_length  # linear activation:
+                )
+            elif block_type == "S":
+                num_coefficients = (
+                    int((prediction_length / 2) - 1) + 1
+                )  # according to paper, dont use floor because prediction_length=1 should be mapped to 0
+                if has_backcast:
+                    self.theta_backcast = mx.gluon.nn.Dense(
+                        units=2 * num_coefficients  # linear activation:
+                    )
+                    self.backcast = mx.gluon.nn.HybridLambda(
```

Comment:
In my idea, I think this would just be a method of the subclass in the interpretable modes (which just does the `F.dot`) or a dense layer in the generic mode.